DSharpPlus.Analyzers/DSharpPlus.Analyzers/Core/SingleEntityGetRequestAnalyzer.cs

```diff
@@ -0,0 +1,122 @@
+using System.Collections.Generic;
+using System.Collections.Immutable;
+using Microsoft.CodeAnalysis;
+using Microsoft.CodeAnalysis.CSharp;
+using Microsoft.CodeAnalysis.CSharp.Syntax;
+using Microsoft.CodeAnalysis.Diagnostics;
+
+namespace DSharpPlus.Analyzers.Core;
+
+[DiagnosticAnalyzer(LanguageNames.CSharp)]
+public class SingleEntityGetRequestAnalyzer : DiagnosticAnalyzer
+{
+    public const string DiagnosticId = "DSP0007";
+    public const string Category = "Design";
+
+    private static readonly LocalizableString title = new LocalizableResourceString(
+        nameof(Resources.DSP0007Title),
+        Resources.ResourceManager,
+        typeof(Resources)
+    );
+
+    private static readonly LocalizableString description = new LocalizableResourceString(
+        nameof(Resources.DSP0007Description),
+        Resources.ResourceManager,
+        typeof(Resources)
+    );
+
+    private static readonly LocalizableString messageFormat = new LocalizableResourceString(
+        nameof(Resources.DSP0007MessageFormat),
+        Resources.ResourceManager,
+        typeof(Resources)
+    );
+
+    private static readonly DiagnosticDescriptor rule = new(
+        DiagnosticId,
+        title,
+        messageFormat,
+        Category,
+        DiagnosticSeverity.Info,
+        true,
+        description
+    );
+
+    public override ImmutableArray<DiagnosticDescriptor> SupportedDiagnostics { get; } = ImmutableArray.Create(rule);
+
+    private static readonly IReadOnlyDictionary<string, string> methods = new Dictionary<string, string>()
+    {
+        { "GetMessageAsync", "DSharpPlus.Entities.DiscordChannel" }
+    };
+
+    private static readonly IReadOnlyDictionary<string, string> preferedMethods = new Dictionary<string, string>()
+    {
+        { "GetMessageAsync", "GetMessagesAsync" }
+    };
```

Comment:
the other relevant methods should be added to this list

Reply:
I added the only ones that seems realistic to have here. Don't know if I missed any other ones